applications/tasks/src/modules/kt_users.erl

```diff
@@ -0,0 +1,556 @@
+%%%-----------------------------------------------------------------------------
+%%% @copyright (C) 2016-2018, Voxter
+%%% @doc
+%%% @author Ben Bradford
+%%% @end
+%%%-----------------------------------------------------------------------------
+-module(kt_users).
+
+%% behaviour: tasks_provider
+
+-export([init/0
+        ,help/1, help/2, help/3
+        ,output_header/1
+        ,cleanup/2
+        ]).
+
+%% Verifiers
+-export([verify_account_id/1
+        ,verify_user_id/1
+        ,verify_email/1
+        ]).
+
+%% Appliers
+-export([import/3
+        ,delete/3
+        ]).
+
+-include("tasks.hrl").
+-include_lib("kazoo_tasks/include/task_fields.hrl").
+
+-define(LIST_BY_USERNAME, <<"users/list_by_username">>).
+
+-define(MATCH_USER_ID(UserId)
+       ,<<(UserId):32/binary>>
+       ).
+
+-define(MOD_CAT, <<(?CONFIG_CAT)/binary, ".users">>).
+-define(DB_DUMP_BULK_SIZE
+       ,kapps_config:get_integer(?MOD_CAT, <<"db_page_size">>, 1000)
+       ).
+
+
+-define(CATEGORY, "users").
+-define(ACTIONS, [<<"import">>
+                 ,<<"delete">>
+                 ]).
+
+%%%=============================================================================
+%%% API
+%%%=============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec init() -> 'ok'.
+init() ->
+    _ = tasks_bindings:bind(<<"tasks.help">>, ?MODULE, 'help'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".output_header">>, ?MODULE, 'output_header'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".cleanup">>, ?MODULE, 'cleanup'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".account_id">>, ?MODULE, 'verify_account_id'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".email">>, ?MODULE, 'verify_email'),
+    _ = tasks_bindings:bind(<<"tasks."?CATEGORY".user_id">>, ?MODULE, 'verify_user_id'),
+    tasks_bindings:bind_actions(<<"tasks."?CATEGORY>>, ?MODULE, ?ACTIONS).
+
+-spec output_header(kz_term:ne_binary()) -> kz_tasks:output_header().
+output_header(?NE_BINARY) ->
+    result_output_header().
+
+-spec result_output_header() -> kz_tasks:output_header().
+result_output_header() ->
+    {'replace', output_header() ++ [?OUTPUT_CSV_HEADER_ERROR]}.
+
+-spec output_header() -> kz_tasks:output_header().
+output_header() ->
+    [<<"id">>
+    ,<<"username">>
+    ,<<"first_name">>
+    ,<<"last_name">>
+    ,<<"account_id">>
+    ].
+
+-spec mandatory_import_fields() -> kz_type:proplist().
+mandatory_import_fields() ->
+    [<<"account_id">>
```

Comment:
I wonder if you should get the list of "required" keys off the schema and prepend the non-required ones you want? Like:

```
{ok, UserSchema} = kz_json_schema:load(<<"users">>),
[<<"password">>
 | kz_json:get_list_value(<<"required">>, UserSchema)
].
```

Something along those lines, so that the task will require the fields of the schema.